x/liquidity/keeper/migration_test.go

```diff
@@ -0,0 +1,197 @@
+	require.True(t, found)
+	msgs := simapp.LiquidityKeeper.GetAllPoolBatchDepositMsgs(ctx, poolBatch)
+	require.Equal(t, 1, len(msgs))
+
+	err = simapp.LiquidityKeeper.ExecuteDeposit(ctx, msgs[0], poolBatch)
+	require.NoError(t, err)
+
+	liquidity.EndBlocker(ctx, simapp.LiquidityKeeper)
+	liquidity.BeginBlocker(ctx, simapp.LiquidityKeeper)
+
+	sendCoin := creatorBalance
+	sendCoin.Amount = sendCoin.Amount.QuoRaw(2)
+	simapp.BankKeeper.SendCoins(ctx, poolCreator, derivedAcc, sdk.Coins{sendCoin})
+	simapp.BankKeeper.SendCoins(ctx, poolCreator, smallHolder, sdk.Coins{sdk.NewCoin(pool.PoolCoinDenom, sdk.NewInt(100))})
+	simapp.BankKeeper.SendCoins(ctx, poolCreator, smallHolder2, sdk.Coins{sdk.NewCoin(pool.PoolCoinDenom, sdk.NewInt(99))})
+	simapp.BankKeeper.SendCoins(ctx, poolCreator, smallHolder3, sdk.Coins{sdk.NewCoin(pool.PoolCoinDenom, sdk.NewInt(1))})
+
+	beforeCommunityFund := simapp.DistrKeeper.GetFeePoolCommunityCoins(ctx)
+
+	err = keeper.SafeForceWithdrawal(ctx, simapp.LiquidityKeeper, simapp.BankKeeper, simapp.AccountKeeper)
+	require.NoError(t, err)
+
+	require.Equal(t, "100denomX,100denomY", simapp.BankKeeper.GetAllBalances(ctx, smallHolder).String())
+	require.Equal(t, "99denomX,99denomY", simapp.BankKeeper.GetAllBalances(ctx, smallHolder2).String())
+	require.Equal(t, "1denomX,1denomY", simapp.BankKeeper.GetAllBalances(ctx, smallHolder3).String())
+	require.Equal(t, "499800denomX,499800denomY", simapp.BankKeeper.GetAllBalances(ctx, poolCreator).String())
+	require.Equal(t, "1000000000000000000000000denomX,1000000000000000000000000denomY", simapp.BankKeeper.GetAllBalances(ctx, depositer).String())
+	require.Equal(t, "500000poolD35A0CC16EE598F90B044CE296A405BA9C381E38837599D96F2F70C2F02A23A4", simapp.BankKeeper.GetAllBalances(ctx, derivedAcc).String())
+
+	afterCommunityFund := simapp.DistrKeeper.GetFeePoolCommunityCoins(ctx)
+	require.EqualValues(t, beforeCommunityFund.Add(sdk.NewDecCoins(sdk.NewDecCoin(denomX, sdk.NewInt(500000)), sdk.NewDecCoin(denomY, sdk.NewInt(500000)))...), afterCommunityFund)
+
+	withdrawMsg := types.NewMsgWithdrawWithinBatch(derivedAcc, 1, sdk.NewCoin(pool.PoolCoinDenom, sdk.NewInt(500000)))
+	_, err = simapp.LiquidityKeeper.WithdrawWithinBatch(ctx, withdrawMsg)
+	require.ErrorIs(t, err, types.ErrPoolNotExists)
+	require.Equal(t, "500000poolD35A0CC16EE598F90B044CE296A405BA9C381E38837599D96F2F70C2F02A23A4", simapp.BankKeeper.GetAllBalances(ctx, derivedAcc).String())
+
+	require.Len(t, simapp.LiquidityKeeper.GetAllDepositMsgStates(ctx), 0)
+	require.Len(t, simapp.LiquidityKeeper.GetAllSwapMsgStates(ctx), 0)
+	require.Len(t, simapp.LiquidityKeeper.GetAllWithdrawMsgStates(ctx), 0)
+	require.Len(t, simapp.LiquidityKeeper.GetAllPoolBatches(ctx), 0)
+	require.Len(t, simapp.LiquidityKeeper.GetAllPools(ctx), 0)
+
+	err = keeper.SafeForceWithdrawal(ctx, simapp.LiquidityKeeper, simapp.BankKeeper, simapp.AccountKeeper)
+	require.NoError(t, err)
+}
```

Comment:
For consistency, an assertion can be added to verify that the pool’s reserve account is emptied after executing `SafeForceWithdrawal()` without errors.